M2/Macaulay2/m2/help.m2

```diff
@@ -501,33 +501,25 @@ setAttribute(infoHelp#0, ReverseDictionary, symbol infoHelp)
 -- View brief documentation within Macaulay2 using symbol?
 -----------------------------------------------------------------------------
 -- TODO: should this return a hypertext object instead of printing?
-briefDocumentation = method(Dispatch => Thing)
-briefDocumentation VisibleList := key -> null
-briefDocumentation Thing       := key -> (
-    if not isDocumentableThing key
-    then return if hasAttribute(key, ReverseDictionary) then (
-	S := getAttribute(key, ReverseDictionary);
+briefDocumentation = key -> (
+    if not isDocumentableMethod key and not isDocumentableThing key
+    then return if hasAttribute(key, ReverseDictionary) then DIV {
 	-- TODO: use either "formation" to enhance the result
 	-- or enhance "describe" or "getTechnical" using "formation"
-	DIV {
-	    BinaryOperation{symbol :=, key, describe key},
-	    getTechnical(S, key)
-	    }
-	);
+	-- TODO: add spaces around :=
+	BinaryOperation{symbol :=, key, describe key},
+	getTechnical(getAttribute(key, ReverseDictionary), key) };
     rawdoc := fetchAnyRawDocumentation makeDocumentTag key;
     -- TODO: should it be getGlobalSymbol or getAttribute?
-    symb := getGlobalSymbol toString key;
     tag := getOption(rawdoc, symbol DocumentTag);
     title := getOption(rawdoc, Headline);
     synopsis := getSynopsis(key, tag, rawdoc);
-    waystouse := documentationValue(symb, key);
-    technical := getTechnical(symb, key);
-    DIV {
-    if title     =!= null then PARA {key, commentize title},
-    synopsis,
-    waystouse,
-    technical }
-)
+    try symb := getGlobalSymbol toString key then (
+	waystouse := documentationValue(symb, key);
+	technical := getTechnical(symb, key)) else ();
+    DIV nonnull {
+	PARA {format tag, commentize title},
+	synopsis, waystouse, technical })
```

Comment:
What happens here if we end up in the `else` statement above and `waystouse` and `technical` are undefined?

Reply:
`nonnull` gets rid of them.

Reply:
Okay, I think maybe I don't completely understand how local variables work.  I would have expected that `waystouse` and `technical` would be symbols if the call to `getGlobalSymbol` raised an error.  But they're `null` instead?

The following seems to indicate that that's the case:

```m2
i1 : f = () -> if false then x := 2 else x

o1 = f

o1 : FunctionClosure

i2 : f() === null

o2 = true
```

Reply:
I think when the function is parsed, every instance of `:=` causes a new local variable with initial value null to be created in the function closure. This is a part of the interpreter that I don't fully understand because it's not documented and the source is very readable either. For instance, this is wild to me:
```m2
i1 : f = x -> y -> (print x; x = y);

i2 : g = f(2);

i3 : g(3)
2

o3 = 3

i4 : g(4)
3 -- it's changed!!!

o4 = 4
```

Reply:
Ah, ok -- interesting